Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Aging penalty fix

## Changelog:
- removed vanilla mortality system, so all became immortal (in `00_defines.txt` file)
- instead of vanilla system we have `character modifiers`, which reduce `health`/`prowess`
- It uses equivalent human age. We apply modifier after >= 30(human) years old. It is updated on every birthday. There are 365 days, and each character has it own birthday, so it doesnt touch performance
- we have two variations of that penalty. One with prowess, one without prowess. It depends on `dynasty legacy` (which removes prowess reduction)
- removed `game start health` bonus, i believe it was used to make them live longer
<details>
<summary>Looks like this</summary>

![image](https://user-images.githubusercontent.com/84410981/231363037-dd8a0b26-52bd-4520-aae3-e067668287cc.png)
![image](https://user-images.githubusercontent.com/84410981/231673803-689c9049-45a5-4dda-aaa3-6f99519b480b.png)

</details>


## Data for health penalty calculation:
I wrote vanilla numbers in excel to calculate how fast it reduces health/prowess with vanilla mortality system. Use `10.0 health` and `100 prowess` as starting numbers.
<details>
<summary>Age 25-80</summary>

![image](https://user-images.githubusercontent.com/84410981/231354363-352f36c3-11b1-4c2a-8f7d-39d1eba00e79.png)
</details>
<details>
<summary>Age 80-120</summary>

![image](https://user-images.githubusercontent.com/84410981/231384154-2ef9f7be-3bd3-4af8-a0b4-533698ed4064.png)
</details>

- I increased prowess losing chance number a little bit, because vanilla uses preudorandom
- In `modifier file` i added to every modifier comments like `#+0.2`. It means i make health penalty bigger from above excel table by this amount. It is required because game thinks it is illness, so it adds health resistance bonus when detect modifier.

## To-do-list:
- [x] Balance aging penalties, so humanity dies like in vanilla (health resistance bonus is main enemy)
## Tests:
- [x] There are no errors in `changed files` in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log`


# How to test:
1. Start up version `1.8.2` of **Crusader Kings III** and look for human characters using _observer mode_.
2. If human dies as intended, you can check other races too, but they should be ok as long as human race is ok.
